drivers/soundwire/master.c

```diff
@@ -84,6 +84,18 @@ int sdw_master_device_add(struct sdw_master_device *md,
 		}
 	}
 
+	/* the bus handle may be set initially or in the link_ops, check now */
+	if (!md->bus)
+		goto link_add_err;
```

Comment:
just to check, that this is really intended: if `nd->bus` is `NULL` here, we return 0 and `sdw_add_bus_master()` is never called?

Reply:
no, nice catch. We should set ret = -EINVAL here. thanks for spotting this.